### ISSUE TITLE
Support for .mdx files

### ISSUE THREAD

**Issue by malouro** [thlorenz/doctoc#211]
Hey all, we'd like to use this tool to generate TOCs in a bunch of [`.mdx`](https://mdxjs.com/) files, though it seems like out-of-the-box it only supports `.md` and `.markdown`. How does adding `.mdx` to the list of valid markdown extensions sound? Or, alternatively, a way to pass in user-defined file extension(s) via some CLI flag? (`--ext=mdx` ?)

In the meantime, as a workaround, we've been using something like `find . -name "*.mdx" | xargs doctoc ...`, though it'd be nice to have it handled & supported via `doctoc` itself.

Thanks!

**Comment by AndrewSouthpaw** [comment]
Hey @malouro, definitely no objections! `.mdx` seems enough like in the markdown family to include it.

Since we advertise as editing _markdown_ files, let's start with passing an option like `--ext=md`, that seems like a good first step.

**Comment by chris-dura** [comment]
FWIW, I'm actually running this tool on `.mdx` files (for Storybook stories) and it seems to work...

```
doctoc src/stories/**/*.mdx src/components/**/*.mdx --gitlab --maxlevel 3 --notitle
```

The "problem" I have is that the `<!-- START | END -->` comments break my highlighting in VS Code because `.mdx` files don't support HTML style comments.

So, I wonder if there can be a configuration option to change the `commentStyle` from HTML to JSX like `{*\ START | END /*}`?

**Comment by I-3B** [comment]
@AndrewSouthpaw is this still open for contribution?

**Comment by AndrewSouthpaw** [comment]
Yep I can review contributions, just tag me as a reviewer and @ me.
